analyzer.go

```diff
@@ -94,6 +94,20 @@ func (a *Analyzer) skipCodeGen(file *ast.File) ([]*ast.CommentGroup, []*ast.Comm
 		}
 	}
 
+	for len(list) > 0 {
+		if strings.Contains(list[0].Text, "go:") || strings.Contains(list[0].Text, "+build") {
```

Comment:
This is still a naive implementation; not all Go directives start with `go:`.

Reply:
Currently, I see only the option with adding all known directives for this issue. 

Do you see any other solution?

Reply:
Yes, there are several possibilities:
1. The directives are excluded when getting the comment, so a "diff" can be used.
2. The directives syntax follows a strict format, so this format can be used.

Reply:
Those options sound acceptable, but could you provide any examples? Those options sound acceptable, but could you provide any examples?

I found that staff like //extern //export //line and other things are internal. So we don't have any options except just adding them manually. 

See at https://github.com/golang/go/blob/master/src/cmd/compile/internal/syntax/scanner.go#L807

And https://github.com/golang/go/blob/master/src/cmd/cgo/ast.go#L291

So, I am going to just add a common default directives and an option to add more directives to skip.

Reply:
The directives syntax follows a strict format: `^[a-z]+:[a-z]+`

https://groups.google.com/g/golang-dev/c/r4rdPdsH1Fg/m/tNZazPenX5cJ?pli=1

`line`,`export`, etc. are not internals: they are parsed inside code from `internal` packages, but they are not internals.

https://github.com/golang/go/blob/d184f8dc020ac635cea02c046ab1d0b87dfd624d/src/go/printer/comment.go#L110-L138